modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/impl/TaskActivityServiceImpl.java

```diff
@@ -30,64 +37,120 @@ public TaskActivityServiceImpl(TaskActivitiesDataService taskActivitiesDataServi
     }
 
     @Override
-    public void addError(Task task, TaskHandlerException e, Map<String, Object> parameters) {
-        taskActivitiesDataService.create(new TaskActivity(e.getMessage(), e.getArgs(), task.getId(),
-                TaskActivityType.ERROR, ExceptionUtils.getStackTrace(e), parameters));
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
We need to have the activity commited, so it can be used later on to update the status of the task execution.